File: C#_Databases_Fundamentals/Exercises/03. EntityFramework Code-First/CodeFirstEF/Hotel.Models/Employee.cs
namespace Hotel.Models
{
    using System.ComponentModel.DataAnnotations;

    public class Employee
    {
        [Key]
        public int Id { get; set; }

        [Required]
        public string FirstName { get; set; }

        [Required]
        public string LastName { get; set; }

        [Required]
        public string Title { get; set; }

        public string Notes { get; set; }
    }
}

File: Java_Fundamentals_January_2017/OOP_Basics/Exersices/03. Inheritance/src/CompanyHierarchy/Models/Employee.java
package CompanyHierarchy.Models;

import CompanyHierarchy.Interfaces.IEmployee;
import CompanyHierarchy.Enums.Department;

public class Employee extends Person implements IEmployee {
    private double salary;
    private Department department;

    public Employee (int id, String firstName, String lastName, Department department) {
        super(id, firstName, lastName);
        this.setDepartment(department);
    }

    private void setSalary (double salary) {
        if (salary < 0) {
            throw new IllegalArgumentException("Invalid input!");
        }

        this.salary = salary;
    }

    private void setDepartment (Department department) {
        this.department = department;
    }

    @Override
    public double getSalary() {
        return this.salary;
    }

    @Override
    public Department getDepartment() {
        return this.department;
    }

    @Override
    public String toString() {
        return "Employee{" +
                "salary=" + salary +
                ", department=" + department +
                '}';
    }
}
